graphql-lwt/src/graphql_lwt.ml

```diff
@@ -21,11 +21,27 @@ module Server = struct
     Schema.execute schema ctx ~variables doc
 
   let execute_request ctx schema req body =
-    Cohttp_lwt.Body.to_string body >>= fun body' ->
-    Lwt_io.printf "Body: %s\n" body';
-    let json = Yojson.Basic.from_string body' in
-    let query = Yojson.Basic.(json |> Util.member "query" |> Util.to_string) in
-    let variables = try Yojson.Basic.Util.(json |> member "variables" |> to_assoc) with _ -> [] in
+    let uri = Cohttp.Request.uri req in
+    (* Use query params if the "query" query param exists; otherwise use the
+       body of the request *)
+    (match Uri.get_query_param uri "query" with
```

Comment:
What are your thoughts on checking `req.meth` rather than whether `query` is present as a query parameter? I.e. when doing a POST to `/graphql?query=...`.

Reply:
Good point! Would be great to support all the recommended behaviours.